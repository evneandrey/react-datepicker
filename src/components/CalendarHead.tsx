import * as React from 'react';
import SVGIcon from './SVGIcon';

interface Props {
  /** Prev button click event */
  onPrev?: () => void;
  /** Next button click event */
  onNext?: () => void;
  /** Calenar Title Click Event */
  onTitleClick?: () => void;
  /** Prev Icon show or Hide */
  prevIcon?: boolean;
  /** Next icon show or hide */
  nextIcon?: boolean;
  /** Title to show in calendar  */
  title?: string;
}

const defaultProps = {
  title: '',
};

const CalendarHead: React.FunctionComponent<Props> = ({
  onPrev,
  onNext,
  prevIcon,
  nextIcon,
  title,
  onTitleClick,
}) => {
  return (
    <div className="calendar__head">
      <div className="calendar__head--prev">
        {prevIcon && (
          <button onClick={onPrev} className="calendar__head--button">
	          <SVGIcon id="left-arrow"/>
          </button>
        )}
      </div>
      <h2 className="calendar__head--title" onClick={onTitleClick}>
        {title}
      </h2>
      <div className="calendar__head--next">
        {nextIcon && (
          <button onClick={onNext} className="calendar__head--button">
	          <SVGIcon id="right-arrow"/>
          </button>
        )}
      </div>
    </div>
  );
};

CalendarHead.defaultProps = defaultProps;

export default CalendarHead;
