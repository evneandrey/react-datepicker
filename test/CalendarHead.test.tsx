import { mount, shallow, ShallowWrapper } from 'enzyme';
import * as React from 'react';
import * as sinon from 'sinon';
import CalendarHead from '../src/components/CalendarHead';

describe('<CalendarHead/>', () => {
  let shallowComponent: ShallowWrapper<React.Component>;

  it('should render correctly', () => {
    shallowComponent = shallow(<CalendarHead />);

    expect(shallowComponent).toBeTruthy();
    expect(shallowComponent).toMatchSnapshot();
  });

  it('should props title correctly', () => {
    shallowComponent = shallow(<CalendarHead title="2019/01" />);

    expect(shallowComponent).toMatchSnapshot();
    expect(shallowComponent.find('h2.calendar__head--title').text()).toEqual('2019/01');
  });

  it('should props prevIcon, nextIcon correctly', () => {
    shallowComponent = shallow(<CalendarHead prevIcon nextIcon />);

    expect(shallowComponent).toMatchSnapshot();
    expect(shallowComponent.find('div.calendar__head--prev > button')).toHaveLength(1);
    expect(shallowComponent.find('div.calendar__head--next > button')).toHaveLength(1);
  });

  it('should props onPrev, onNext correctly', () => {
    const onPrev = sinon.spy();
    const onNext = sinon.spy();

    shallowComponent = shallow(<CalendarHead prevIcon nextIcon onPrev={onPrev} onNext={onNext} />);

    expect(shallowComponent).toMatchSnapshot();
    shallowComponent.find('div.calendar__head--prev > button').simulate('click');
    shallowComponent.find('div.calendar__head--next > button').simulate('click');

    expect(onPrev).toHaveProperty('callCount', 1);
    expect(onNext).toHaveProperty('callCount', 1);
  });
});
