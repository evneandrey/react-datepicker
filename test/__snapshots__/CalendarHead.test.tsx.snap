// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CalendarHead/> should props onPrev, onNext correctly 1`] = `
<div
  className="calendar__head"
>
  <div
    className="calendar__head--prev"
  >
    <button
      className="calendar__head--button"
      onClick={[Function]}
    >
      <SVGIcon
        color="currentColor"
        id="left-arrow"
        size="16"
      />
    </button>
  </div>
  <h2
    className="calendar__head--title"
  />
  <div
    className="calendar__head--next"
  >
    <button
      className="calendar__head--button"
      onClick={[Function]}
    >
      <SVGIcon
        color="currentColor"
        id="right-arrow"
        size="16"
      />
    </button>
  </div>
</div>
`;

exports[`<CalendarHead/> should props prevIcon, nextIcon correctly 1`] = `
<div
  className="calendar__head"
>
  <div
    className="calendar__head--prev"
  >
    <button
      className="calendar__head--button"
    >
      <SVGIcon
        color="currentColor"
        id="left-arrow"
        size="16"
      />
    </button>
  </div>
  <h2
    className="calendar__head--title"
  />
  <div
    className="calendar__head--next"
  >
    <button
      className="calendar__head--button"
    >
      <SVGIcon
        color="currentColor"
        id="right-arrow"
        size="16"
      />
    </button>
  </div>
</div>
`;

exports[`<CalendarHead/> should props title correctly 1`] = `
<div
  className="calendar__head"
>
  <div
    className="calendar__head--prev"
  />
  <h2
    className="calendar__head--title"
  >
    2019/01
  </h2>
  <div
    className="calendar__head--next"
  />
</div>
`;

exports[`<CalendarHead/> should render correctly 1`] = `
<div
  className="calendar__head"
>
  <div
    className="calendar__head--prev"
  />
  <h2
    className="calendar__head--title"
  />
  <div
    className="calendar__head--next"
  />
</div>
`;
